src/components/Table/utils/RowState.ts

```diff
@@ -172,6 +196,24 @@ export class RowState {
     );
   }
 
+  get isLastKeptRow(): boolean {
```

Comment:
I added a few new methods here, b/c it will help short-circuit the row re-rendering; i.e.

* Without this getter, the RowImpl itself will access keptRows, and anytime kept rows changes, all RowImpls will react, which also means re-render
* With this getter, when keptRows changes, all RowStates will re-calc their isLastKeptRow, but nearly all of them will be the same value, so the RowImpl observer watching this getter won't re-run, hence no re-render